raiden/network/rpc/smartcontract_proxy.py

```diff
@@ -129,6 +130,10 @@ def transact(
                         "local transaction pool. Bailing ..."
                     )
 
+                raise EthereumNonceTooLow(
```

Comment:
The [test failure](https://circleci.com/gh/raiden-network/raiden/87817#tests/containers/24) shows we also do not handle this for geth:

> raiden.exceptions.RaidenUnrecoverableError: Unexpected error in underlying Ethereum node: {'code': -32000, 'message': 'nonce too low'}

So I suppose you should handle it and add an assumption test for geth too.